client/src/varclient.c

```diff
@@ -176,8 +179,23 @@ int ClientRequest( VarClient *pVarClient, int signal )
             {
                 do
                 {
-                    pVarClient->blocked = 1;
-                    result = sem_wait( &pVarClient->sem );
+                    if ( pVarClient->requestTimeout_ms > 0) {
+                        result = clock_gettime(CLOCK_REALTIME, &pVarClient->ts);
+                        if( result == -1)
+                        {
+                            continue;
+                        }
+
+                        // Add 5 second timeout
+                        pVarClient->ts.tv_sec += 5;
```

Comment:
Shouldn't you be using requestTimeout_ms?

Reply:
and if you are adding it to tv_sec, then it should be measured in seconds, not milliseconds

Reply:
oh, silly me, forgot to update. Well, I switched to setting timeout in seconds anyway.